src/components/action-popover/action-popover.test.tsx

```diff
@@ -0,0 +1,2303 @@
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    expect(screen.getByRole("link")).toHaveTextContent("test download");
+  });
+
+  it("should close the menu if enter pressed", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem href="#" download>
+          test download
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    await screen.getByRole("link").focus();
+    await user.keyboard("{Enter}");
+
+    expect(screen.queryByRole("list")).not.toBeInTheDocument();
+  });
+});
+
+test("displays the vertical ellipsis icon as the menu button", () => {
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+  expect(screen.getByTestId("icon")).toHaveStyleRule(
+    "content",
+    `"${iconUnicodes.ellipsis_vertical}"`,
+    { modifier: "&::before" }
+  );
+});
+
+test("has expected border radius", () => {
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+  expect(screen.getByRole("button")).toHaveStyle({
+    borderRadius: "var(--borderRadius050)",
+  });
+});
+
+test("has proper data attributes applied to elements", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item 1</ActionPopoverItem>
+      <ActionPopoverDivider />
+      <ActionPopoverItem>example item 2</ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+
+  expect(screen.getByTestId("action-popover-wrapper")).toHaveAttribute(
+    "data-component",
+    "action-popover-wrapper"
+  );
+  expect(screen.getByRole("list")).toHaveAttribute(
+    "data-component",
+    "action-popover"
+  );
+  const divider = screen.getAllByRole("listitem")[1];
+  expect(divider).toHaveAttribute("data-element", "action-popover-divider");
+});
+
+test("has a default aria-label", () => {
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+  expect(screen.getByRole("button")).toHaveAttribute("aria-label", "actions");
+});
+
+test("uses the aria-label prop if provided", () => {
+  render(
+    <ActionPopover aria-label="test aria label">
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+  expect(screen.getByRole("button")).toHaveAttribute(
+    "aria-label",
+    "test aria label"
+  );
+});
+
+test("renders with the menu closed by default", () => {
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+  expect(screen.queryByRole("list")).not.toBeInTheDocument();
+});
+
+test.each<["top" | "bottom", boolean, string]>([
```

Comment:
It wasn't clear to me that this is tested in Playwright. I've just looked again and there are some similar tests, but I'm not sure they're doing exactly the same thing. In particular, I can't see any tests of the `top-start` case.

Obviously though this is a pretty janky test that is hard to read and only tests implementation details, so I'd be more than happy to get rid of it - which on balance I think you're right that I can (unless needed for coverage - I doubt it is but will check).

Reply:
I just looked again - there's definitely no Playwright tests for at least some of the prop combinations tested here. In any case, even the PW tests aren't that good in the sense they're still testing implementation details (`data-floating-placement` attributes, which I believe are added by `floatingui` rather than us anyway) so I don't have much more confidence in those than these janky unit tests. So I'd prefer to leave them for now, although if we later review our tests and decide to move these to Playwright that would be fine.